src/main/scala/scalaz/analytics/LocalAnalyticsModule.scala

```diff
@@ -53,8 +53,8 @@ trait LocalAnalyticsModule extends AnalyticsModule {
   implicit override val byteType: Type[scala.Byte]            = LocalType(Reified.Byte)
   implicit override val nullType: Type[scala.Null]            = LocalType(Reified.Null)
   implicit override val shortType: Type[scala.Short]          = LocalType(Reified.Short)
-  implicit override val timestampType: Type[LocalDateTime]    = LocalType(Reified.Timestamp)
-  implicit override val dateType: Type[LocalDate]             = LocalType(Reified.Date)
+  implicit override val instantType: Type[Instant]            = LocalType(Reified.Instant)
```

Comment:
The more I think about this the more I think we might need a few more types here... Even between these two types here:
- `Instant` is a timestamp with a timezone (offset from epoch)
- `LocalDateTime` is a timestamp without a timezone
Both seem to have plenty of valid use cases.

This is actually one of the consequences of controlling the language users can define - we have to reimplement a *lot* of functionality. In this case seemingly `java.time.*`.

@jdegoes How did you envisage this looking?

Reply:
managing an offset from the epoch (like Instant does) is simple and gets you quite far. as soon as you are starting to think about things like timezones or calendars, you're in deep shit - don't go there if you can avoid it.

even LocalDateTime is tricky - who said _your_ country used the gregorian calendar at all at that date, for example?

Reply:
Yes so using `Instant` is the same strategy as the Impala Timestamp. Just storing an offset with no timezone, and interpreting it relative to UTC. You would be free to store the Timezone separately.

@ritschwumm agree about `LocalDate` but I'd argue the general usefulness outweighs that as an edge case.

Reply:
in one of my own projects i quite successfully used an offset variant of julian days (just counting solar days from some arbitrary day in the past, in my case 1.1.1970) for local dates instead - this has the great advantage of being completely independent of any calendar system.